jujugui/static/gui/src/app/app.js

```diff
@@ -746,7 +746,9 @@ YUI.add('juju-gui', function(Y) {
       // nextProps on the lifecycle methods.
       ReactDOM.render(
         <window.juju.components.UserProfile
+          addNotification={this.db.notifications.add.bind(this)}
           currentModel={this.get('jujuEnvUUID')}
+          destroyModel={this.env.destroyModel.bind(this.env)}
```

Comment:
I followed the pattern for listModels below, I can modify both these calls to use the passed in env if you prefer...